### ISSUE TITLE
inputsourceschange event in XRSession isn't dispatched

### ISSUE THREAD

**Issue by soadzoor** [immersive-web/webxr-polyfill#37]
Hi,

I created a quick mockup about the issue in the title, you can find it here: https://vrtual.house/development/webxr-polyfill-test/

The problem:
I cannot make any interaction on the scene, when the WebXR flag is disabled, and the polyfill is used. In three.js, there are some events for interaction, like `selectstart`, and `selectend`, but they're not dispatched when the polyfill is used, only when the flag is set to enabled.

I guess it has something to do with the fact that the `inputsourceschange` isn't fired either when the polyfill is used. (You can find the lines in question starting from line `21752` in the three.js r96 build):
![capture](https://user-images.githubusercontent.com/10392261/45387837-94197700-b617-11e8-8645-f3c5073274f7.PNG)

**Comment by jsantell** [comment]
Looks like this is not yet implemented in the polyfill. I see `inputsourceschange` in the explainer, but not the actual spec -- @toji should this be implemented here?

**Comment by jsantell** [comment]
So this event should be added to the spec, submitted: https://github.com/immersive-web/webxr/pull/395

**Comment by takahirox** [comment]
I want to go with `inputsourceschange` event support for #81

**Comment by takahirox** [comment]
I'm thinking how to implement. As I mentioned in #81, inputSources are set up in animation frame (more precisely, `onFrameStart()`). So, perhaps easiest solution may be replacing 
 
```javascript
this[PRIVATE].device.onFrameStart(this[PRIVATE].id);
callback(now(), this[PRIVATE].frame);
this[PRIVATE].device.onFrameEnd(this[PRIVATE].id);
```

https://github.com/immersive-web/webxr-polyfill/blob/master/src/api/XRSession.js#L295-L297

with something like this

```javascript
this[PRIVATE].device.onFrameStart(this[PRIVATE].id);

const newInputSources = this.inputSources;
const previousInputSources = this[PRIVATE].previousInputSources;

// Here, compare inputSources with previous frame's ones
// and fire inputsourceschange event if different.

this[PRIVATE].previousInputSources = newInputSources;

callback(now(), this[PRIVATE].frame);
this[PRIVATE].device.onFrameEnd(this[PRIVATE].id);
```

What do you think?

**Comment by jsantell** [comment]
SGTM!

**Comment by takahirox** [comment]
I think this issue can be closed because of #91

**Comment by toji** [comment]
Agreed.
